argo/src/Argo.hs

```diff
@@ -198,8 +204,12 @@ instance JSON.ToJSON JSONRPCException where
           Just theID -> JSON.toJSON theID
       , "error" .=
         (JSON.object $ ["code" .= JSON.Number (fromInteger (errorCode exn))
-                       , "message" .= JSON.String (message exn)] ++
-                       (("data" .=) <$> maybeToList (errorData exn)))
+                      , "message" .= JSON.String (message exn)
+                      , "data" .= (JSON.object $
+                        ([ "stdout" .= errorStdOut exn
+                         , "stderr" .= errorStdErr exn ] ++
+                         (("data" .=) <$> maybeToList (errorData exn))))
```

Comment:
Hmm, the difficulty coming up with a fitting name strikes me as a bit of a protocol wart. Perhaps we should just say that apps always return objects, and then merge in these two extra fields, clobbering any that the app made with those names?

Reply:
I have a slight preference for nesting the field, since it means the application cannot possibly interfere with the field names. However, perhaps we could just name the two extra fields in a sufficiently obfuscated way that it would take intentional subversion to clash with them: for example, `__argo_method_stdout__` or some such. Do you think we need to worry that the client might clash with field names and create difficult-to-debug behavior?

Reply:
I don't think obfuscating the names is that important. If we document that we'll clobber any fields with those names that the app developer inserted, then I don't think they'll be too surprised.

Putting myself in the shoes of an app developer who has seen those two fields in all of my tests, I would not be frustrated by them overwriting my own that I later inserted. We could also throw an exception if the developer included them that made it clear what had happened.

Reply:
I ended up resolving the issue thusly: `result` has 4 fields, `state`, `answer`, `stdout`, and `stderr`. This was made possible by making state tracking non-optional in #82.